doc/qos/tunnel_dscp_remapping.md

```diff
@@ -0,0 +1,202 @@
+# DSCP remapping for tunnel traffic
+
+## 1 Table of Content ###
+
+- [Revision](#11-revision)
+- [Scope](#2-scope)
+- [Definitions/Abbreviations](#3-definitionsabbreviations)
+- [Overview](#4-overview)
+- [Design](#5-design)
+    - [SWSS Schema](#51-swss-schema)
+        - [Define new table for mapping](#511-define-new-table-for-mapping)
+        - [Update existing TUNNEL table](#512-update-existing-tunnel-table)
+        - [Define new field for extra lossless queues](#513-define-new-field-for-extra-lossless-queues)
+    - [SAI attribute](#52-sai-attribute)
+    - [orchagent](#53-orchagent)
+- [Test requirement](#6-test-requirement)
+- [Open Questions](#7-open-questions)
+
+### 1.1 Revision ###
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 |             | Bing Wang   | Initial version                   |
+
+
+## 2 Scope ##
+
+This document covers high level design of DSCP and TC remapping for tunnel traffic in SONiC.
+
+## 3 Definitions/Abbreviations ##
+
+
+| Term | Meaning |
+|:--------:|:---------------------------------------------:|
+| PFC | Priority-based Flow Control  |
+| TC | Traffic class|
+| DSCP| Differentiated Services Code Point |
+
+## 4 Overview
+
+The current QoS map architecture allows for port-based selection of each QoS map. However, we are not able to override the port-based QoS map for tunnel traffic. 
+This design proposes a method to remapping DSCP and TC for IPinIP tunnel. 
+
+
+## 5 Design ##
+
+### 5.1 SWSS Schema
+#### 5.1.1 Define new table for mapping
+Please be noted that below config is to remap traffic in queue 3 to queue 2, and traffic in queue 4 to queue 6.
+* Table for decap
+
+    DSCP_TO_TC_MAP for mapping DSCP to TC
+    ```json
+    "DSCP_TO_TC_MAP": {
```

Comment:
How are these tables going to get added to the config db? We need to mention that as well

Reply:
Updated. Thanks

Reply:
It's a good finding. I will take a closer look.

Reply:
Please list all the dscp to tc mappings here

Reply:
Updated. Thanks

Reply:
> Just FYI. Currently, there is a logic in [QoSOrch](https://github.com/Azure/sonic-swss/blob/master/orchagent/qosorch.cpp#L284) that applies the DSCP_TO_TC_MAP as a global attribute. Currently, there is only one object in the DSCP_TO_TC_MAP table so it should be fine. But I'm not sure whether it will be problematic if there are more than one.

@bingwang-ms @neethajohn 
Is this issue addressed? 
At the SAI level, looks like the map will be applied to all ports when this SWITCH attribute is set.
At the SONiC level, it always applies to SWITCH once a new `DSCP_TO_TC_MAP` is configured, which means the one configured last will always take effect.
There is no guarantee that which map will be applied first and which is second. In case the one for tunnel decapsulation is applied last, it will override the map of all ports.
We may want to add a configuration knob to indicate which one should be configured to SWITCH level.